ultralytics/solutions/dwell_time_analyzer.py

```diff
@@ -0,0 +1,475 @@
+                    box_style=True,
+                )
+            offset += 30
+
+        # Display average dwell times if enabled
+        if self.enable_avg_dwell and avg_dwell_times:
+            annotator.text(
+                (10, offset),
+                "Average Dwell Times (per zone):",
+                txt_color=self.analytics_text_color,
+                box_style=True,
+            )
+            offset += 30
+            for z_name, avg_dt in avg_dwell_times.items():
+                unit = "s" if self.fps else "frames"
+                annotator.text(
+                    (10, offset),
+                    f"  {z_name}: {avg_dt:.2f}{unit}",
+                    txt_color=self.analytics_text_color,
+                    box_style=True,
+                )
+                offset += 30
+
+        im0[:] = annotator.result()
+
+    def draw_zones(self, im0: Any, current_zone_counts: Dict[str, int]) -> None:
+        """
+        Draws the defined zones on the image frame along with their labels and current counts.
+
+        Args:
+            im0 (Any): The image frame on which to draw the zones.
+            current_zone_counts (Dict[str, int]): Current counts of objects in each zone.
+        """
+        annotator = Annotator(im0, line_width=self.line_width)
+        for zone_name, polygon in self.zone_polygons.items():
+            if polygon.geom_type == "Polygon":
+                coords = list(polygon.exterior.coords)
+            else:
+                coords = list(polygon.coords)
+            coords = [(int(x), int(y)) for (x, y) in coords]
+            # Blue color for the zone
+            annotator.draw_region(
+                reg_pts=coords,
+                color=(255, 0, 0),  # BGR for Blue
+                thickness=self.line_width,
+            )
+            # Red color for the zone label
+            count_in_zone = current_zone_counts.get(zone_name, 0)
+            zone_label = f"{zone_name} | Count: {count_in_zone}"
+            if coords:
+                annotator.text(
+                    (coords[0][0], coords[0][1] - 10),
+                    zone_label,
+                    txt_color=(0, 0, 255),
+                    box_style=True,
+                )  # Red label
+        im0[:] = annotator.result()
+
+    def count(self, im0: Any) -> Any:
+        """
+        Processes each frame to track objects, update dwell times, and display analytics.
+
+        Args:
+            im0 (Any): The current image frame to process.
+
+        Returns:
+            Any: The annotated image frame.
+        """
+        self.frame_number += 1
+        # Pass classes to model.track if classes are specified
+        self.tracks = self.model.track(source=im0, persist=True, classes=self.CFG.get("classes"), **self.track_add_args)
+
+        # Extract tracks for OBB or object detection
+        track_data = self.tracks[0].obb or self.tracks[0].boxes
+
+        if track_data and track_data.id is not None:
```

Comment:
@Muhammad-237 This code is already handled by solutions class: https://github.com/ultralytics/ultralytics/blob/a3d807be13af2d8fa008f522020411d2d37e4860/ultralytics/solutions/solutions.py#L93

Can you please use the solutions call directly, Thank you!

Reply:
this issue has been solved